### PR TITLE
[CDAP-8497] In new UI, show appropriate message when system services are down

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-8497

- Added a poll to `/v3/system/services/status` to check for system services status, in addition to the namespace check that we're doing right now.
- When either poll returns an error, show 'CDAP Services are not available'.

Also changed `/backendstatus`endpoint to `/namespacestatus`, since the code at that endpoint is just checking for namespaces, not actually doing all the checks required to know whether or not the backend is down. The corresponding old UI change for this is at: fce325ffe09f93a1039a9c5f5d6c6add7def7a5a

### CODE REVIEW

**Comment by ajainarayanan** [comment]
Talked to @tonybach. Two things,
- He is working on changing the way we check status. We can do this via sockets instead of http
- Merge conflicts.

**Comment by ajainarayanan** [comment]
Talked to @tonybach. Two things,
1. When CDAP backend is down or is being restarted we show "CDAP Services are not available"
   And if the UI server is restarted at this time we switch the message to `User interface service is not available` but when UI comes back up and if the backend is still starting we don't switch the message back to `CDAP Services are not available`.We still still show `User Interface service is down` message.
2. The messaging for `User Interface service is down` should be consistent to the other message we show. It should be `User Interface Service is not available` 

@bdmogal Thoughts?

**Comment by tonybach** [comment]
Talked with @ajainarayanan and @bdmogal, we decided to change the `CDAP Services are not available` to `Unable to communicate with CDAP services`, and `Attempting to connect...` to `Retrying...` with ellipsis animation, to be more consistent with another error message that we had.

Also, will add an additional condition to show the backend down message: when the call to `system/services/status` returns 200, but not all services' status are 'OK'.

**Comment by elfenheart** [comment]
This doesn't seem to work when node server is running but cdap backend is not running. The status down modal is not showing up

File: cdap-ui/app/cdap/services/StatusFactory.js [hunk 1]
```diff
@@ -26,61 +26,69 @@ let pollCycleInProgress = false;
...
+      let [namespaceRes, serviceStatusRes] = responses;
+      serviceStatusRes.json()
+        .then(serviceStatuses => {
+          /*
+            Check for 3 things
+              1. /v3/namespaces returns a non-error response
+              2. /v3/system/services/status returns a non-error response
+              3. All the system services have status of 'OK'
+          */
+          if (namespaceRes.status < 500 && serviceStatusRes.status < 500 && serviceStatuses.body.indexOf('NOTOK') === -1) {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/services/StatusFactory.js]: any particular reason why `< 500`? this could technically be 404 I believe

Comment (tonybach) [reply cdap-ui/app/cdap/services/StatusFactory.js]: I think @bdmogal said that we would never get 400 errors from these endpoints?

**Comment by tonybach** [comment]
Putting on-hold until we figure out what to do with the different retry messages we currently have. In @bdmogal 's words:

```For example, there are cases when you would see a CDAP system services down message in a modal, then once that disappears, you would see the retry message on the overview page.```

**Comment by nitinmotgi** [comment]
Either close this PR or take it to completion. It's been hanging in there for long enough.

**Comment by bdmogal** [comment]
I think we should close this, and come back to it when we have a better, foolproof solution.

**Pull request closed without merging**